internal/awslambda/extension/telemetry/server.go

```diff
@@ -0,0 +1,97 @@
+package telemetry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/sirupsen/logrus"
+)
+
+type RuntimeDoneHook func()
+
+func NoopHook() RuntimeDoneHook {
+	return func() {}
+}
+
+type Server struct {
+	addr string
```

Comment:
Instead of caching the addr here, I think it might make more sense to store the server so you can get its listen addr once it has been started for testing.

Reply:
Something like ?
```
type Server struct {
    s *http.Server
}

func (s *Server) Addr() {
    return s.Addr()
}
```

I don't see much value in storing the server since nothing is called on it, but it could make it a bit more testable.

Reply:
If you wanted to test a full end to end, that storing the server provides the value that you can call the `server.Addr` and it will give you the bound address instead of the address it started with (ie `localhost:0`).

The other way to test this is expose the middleware or Mux and pass that to `httptest.NewServer(...)` and do the full end to end tests there.

Reply:
Ah yep, I've changed it to server already, that makes sense, I've just been using the specific address and didn't know you could start with localhost:0